src/java/org/apache/cassandra/index/sai/utils/PrimaryKey.java

```diff
@@ -163,19 +159,61 @@ public String toString()
             }
         }
 
+        class TokenOnlyPrimaryKey extends AbstractPrimaryKey
+        {
+            private final Token token;
+
+            TokenOnlyPrimaryKey(Token token)
+            {
+                this.token = token;
+            }
+
+            @Override
+            public boolean isTokenOnly()
+            {
+                return true;
+            }
+
+            @Override
+            public Token token()
+            {
+                return token;
+            }
+
+            @Nullable
+            @Override
+            public DecoratedKey partitionKey()
+            {
+                return null;
+            }
+
+            @Nullable
+            @Override
+            public Clustering<?> clustering()
+            {
+                return null;
+            }
```

Comment:
nit: I'd consider an override of `asComparableBytes()` here that throw `UnsupportedOperationException` instead of NPE.